dietpi/dietpi-config

```diff
@@ -822,160 +822,441 @@ A long (or insufficiently manufactured) cable may required a higher boost settin
 
 			fi
 
-		# RPi
-		elif (( $G_HW_MODEL < 10 )); then
+			# RPi
+			elif (( $G_HW_MODEL < 10 )); then
+
+				local framebuffer_x=$(grep -m1 '^[[:blank:]]*framebuffer_width=' /DietPi/config.txt || vcgencmd get_config framebuffer_width)
+				framebuffer_x=${framebuffer_x#*=}; framebuffer_x=${framebuffer_x:-0}
+				local framebuffer_y=$(grep -m1 '^[[:blank:]]*framebuffer_height=' /DietPi/config.txt || vcgencmd get_config framebuffer_height)
+				framebuffer_y=${framebuffer_y#*=}; framebuffer_y=${framebuffer_y:-0}
+				local current_value=$(grep -m1 '^[[:blank:]]*dtoverlay=vc4-' /DietPi/config.txt | sed 's/^[^=]*=//') #OpenGL check 1st
+
+				#If framebuffer is not set by user then check if hdmi default is detected and being used
+				if [[ "$framebuffer_y" == 0 ]] && [[ "$framebuffer_x" == 0 ]]; then
+
+						#Save output of tvservice -s to var | NOTE: tvservice will output a default resolution even if hdmi is Disconnected or unable to pull information from connected device
+						value_of_tvservice_s=$(tvservice -s)
+
+						#Transformation of the output of tvservice -s to get detected or default X,Y resolution
+						#get X
+						tvservice_s_framebuffer_x=${value_of_tvservice_s%x*}  # retain the part before last character x
+						tvservice_s_framebuffer_x=${tvservice_s_framebuffer_x##*,}  # retain the part after the last character ,
+						tvservice_s_framebuffer_x=${tvservice_s_framebuffer_x:1}   # Remove the first character of blank space
+						#get Y
+						tvservice_s_framebuffer_y=${value_of_tvservice_s%@*}  # retain the part before last character @
+						tvservice_s_framebuffer_y=${tvservice_s_framebuffer_y##*x}  # retain the part after the last character x
+						tvservice_s_framebuffer_y=${tvservice_s_framebuffer_y::-1} # Remove the last character of blank space
+
+						# default X,Y resolution to framebuffer vars
+						framebuffer_x=$tvservice_s_framebuffer_x
+						framebuffer_y=$tvservice_s_framebuffer_y
+				fi
 
-			local framebuffer_x=$(grep -m1 '^[[:blank:]]*framebuffer_width=' /DietPi/config.txt || vcgencmd get_config framebuffer_width)
-			framebuffer_x=${framebuffer_x#*=}; framebuffer_x=${framebuffer_x:-0}
-			local framebuffer_y=$(grep -m1 '^[[:blank:]]*framebuffer_height=' /DietPi/config.txt || vcgencmd get_config framebuffer_height)
-			framebuffer_y=${framebuffer_y#*=}; framebuffer_y=${framebuffer_y:-0}
-			local current_value=$(grep -m1 '^[[:blank:]]*dtoverlay=vc4-' /DietPi/config.txt | sed 's/^[^=]*=//') # OpenGL check 1st
-			if [[ ! $current_value ]]; then
+				if [[ ! $current_value ]]; then
 
-				# - FB
-				current_value="$framebuffer_x X $framebuffer_y"
+					# - FB
+					current_value="$framebuffer_x X $framebuffer_y"
 
-				# - Check for headless
-				grep -q '^[[:blank:]]*hdmi_ignore_hotplug=1' /DietPi/config.txt && grep -q '^[[:blank:]]*hdmi_ignore_composite=1' /DietPi/config.txt && current_value='Headless'
```

Comment:
The downsite greps need to be replaced with the upside greps again. This was a recent change in dev code since the missing `^` in downside greps lead to wrong results if those two settings were commented. Also `-i` not required since config.txt is case sensitive.